packages/react-router/modules/matchPath.js

```diff
@@ -26,11 +26,17 @@ const compilePath = (pattern, options) => {
 /**
  * Public API for matching a URL pathname to a path pattern.
  */
-const matchPath = (pathname, path, options = {}) => {
+const matchPath = (pathname, path, options = {}, parentMatch = null) => {
   const { exact = false, strict = false } = options
 
-  if (!path)
-    return { url: pathname, isExact: true, params: {} }
+  const absolute = isAbsolute(path)
+
+  if (path === undefined)
+    return { url: parentMatch ? parentMatch.url : '/', isExact: true, params: {} }
```

Comment:
Using the full pathname breaks relative links inside of the `<Route>`. I'm trying to recall where I ran into an issue with this. Possibly with the `<NavLink>`.

`parentMatch` should only be `null` for routes that have no parent `<Route>`, in which case I believe that `match.url` should be `/` so that any of their children resolve relative to the root.

```js
// location.pathname = /here/is/where/I/am
<Route path='/here' render={() => (
  <Route render={() => (
    // when match.url = pathname, this resolves to /here/is/where/I/there
    // when match.url = parent's match.url, this resolves to /there
    <Link to='../there'>Some Place</Link>
  )} />
)} />
```